src/test/scala/TableGroupTest.scala

```diff
@@ -0,0 +1,395 @@
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.node.{
+  ArrayNode,
+  JsonNodeFactory,
+  ObjectNode
+}
+import org.scalatest.FunSuite
+
+class TableGroupTest extends FunSuite {
+  val objectMapper = new ObjectMapper()
+  test("should create table group object from pre parsed metadata") {
+    val json =
+      """
+        |{
+        |  "@context": "http://www.w3.org/ns/csvw",
+        |  "tables": [{
+        |    "url": "countries.csv",
+        |    "tableSchema": {
+        |      "columns": [{
+        |        "name": "countryCode",
+        |        "titles": "countryCode",
+        |        "datatype": "string",
+        |        "propertyUrl": "http://www.geonames.org/ontology{#_name}"
+        |      }, {
+        |        "name": "latitude",
+        |        "titles": "latitude",
+        |        "datatype": "number"
+        |      }, {
+        |        "name": "longitude",
+        |        "titles": "longitude",
+        |        "datatype": "number"
+        |      }, {
+        |        "name": "name",
+        |        "titles": "name",
+        |        "datatype": "string"
+        |      }],
+        |      "aboutUrl": "http://example.org/countries.csv{#countryCode}",
+        |      "propertyUrl": "http://schema.org/{_name}",
+        |      "primaryKey": "countryCode"
+        |    }
+        |  }, {
+        |    "url": "country_slice.csv",
+        |    "tableSchema": {
+        |      "columns": [{
+        |        "name": "countryRef",
+        |        "titles": "countryRef",
+        |        "valueUrl": "http://example.org/countries.csv{#countryRef}"
+        |      }, {
+        |        "name": "year",
+        |        "titles": "year",
+        |        "datatype": "gYear"
+        |      }, {
+        |        "name": "population",
+        |        "titles": "population",
+        |        "datatype": "integer"
+        |      }],
+        |      "foreignKeys": [{
+        |        "columnReference": "countryRef",
+        |        "reference": {
+        |          "resource": "countries.csv",
+        |          "columnReference": "countryCode"
+        |        }
+        |      }]
+        |    }
+        |  }]
+        |}
+        |""".stripMargin
+    val jsonNode = objectMapper.readTree(json)
+    val tableGroup = TableGroup.fromJson(
+      jsonNode.asInstanceOf[ObjectNode],
+      "http://w3c.github.io/csvw/tests/countries.json"
+    )
+    val table2 =
+      tableGroup.tables("http://w3c.github.io/csvw/tests/country_slice.csv")
+    assert(tableGroup.id.isEmpty)
+    assert(tableGroup.tables.size === 2)
+    assert(tableGroup.annotations.size === 0)
+    assert(table2.foreignKeyReferences.length === 1)
+    assert(
+      table2
+        .foreignKeyReferences(0)
+        .referencedTable
+        .url === "http://w3c.github.io/csvw/tests/countries.csv"
+    )
+    assert(table2.foreignKeyReferences(0).referencedTableColumns.length === 1)
```

Comment:
Please stop repeating tabl2.foreignKeyReferences(0) - stick it in a variable.

And the same in the next unit test please